src/query/interpreter.cpp

```diff
@@ -2425,19 +2594,35 @@ Interpreter::PrepareResult Interpreter::Prepare(const std::string &query_string,
   }
 }
 
+std::vector<std::string> Interpreter::GetQueries() const {
+  if (in_explicit_transaction_) {
+    return {"EXPLICIT TRANSACTION"};
```

Comment:
I also agree that this is not good behaviour, we should return all queries executed in transaction, or have some other way for user to distinguish between transactions. Maybe you want to abort all transactions and that should be fine.

Reply:
@antoniofilipovic why abort all transactions?? I agree that returning all queries when in explicit transaction should be the way to go + that sync between threads is needed. The reason I iterated over `query_executions` vector is because if interpreter runs let's say third query, there will be two nullptr elements in the vector. But yes, I think this can be solved in a nicer way with returning just back() element.

Reply:
But don't you think that returning all queries that are part of explicit transaction can create a mess? E.g when running thousands of queries inside the explicit transaction.

Reply:
I am sorry, I expressed my thoughts incorrectly. I thought about aborting that specific transaction with all the other queries inside.